libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslDiagnostics.java

```diff
@@ -406,4 +438,58 @@ private static boolean checkIssuer(X509Certificate certificate, X509Certificate
     private static boolean isSelfIssued(X509Certificate certificate) {
         return certificate.getIssuerX500Principal().equals(certificate.getSubjectX500Principal());
     }
+
+    private static String keyUsageDescription(X509Certificate certificate) {
+        boolean[] keyUsage = certificate.getKeyUsage();
+        if (keyUsage == null || keyUsage.length == 0) {
+            return "no keyUsage";
+        }
+
+        final String[] keyUsageGlossary = {"digitalSignature", "nonRepudiation", "keyEncipherment",
+            "dataEncipherment", "keyAgreement", "keyCertSign", "cRLSign", "encipherOnly",
+            "decipherOnly"};
+
+        List<String> keyUsageDescription = new ArrayList<>();
+        IntStream.range(0, keyUsage.length).forEach(i -> {
+            if (keyUsage[i]) {
+                keyUsageDescription.add(keyUsageGlossary[i]);
+            }
+        });
+        return keyUsageDescription.stream()
+            .reduce((a, b) -> a + ", " + b)
```

Comment:
The style use have used here is fine, but I would have written it like this:
```suggestion
        String keyUsageDescription = IntStream.range(0, keyUsage.length)
          .filter(i -> keyUsage[i])
          .map(i -> i < keyUsageGlossary.length ? keyUsageGlossary[i] : "#" + i)
          .collect(Collectors.joining(", "));
```
And then tested whether that was empty or not.